### ISSUE TITLE
Idempotence plugin doesn't work

### ISSUE THREAD

**Issue by retr0h** [metacloud/molecule#440]
It doesn't work for 1.9.6 nor does it work with Ansible 2.x.

### 2.0

`runner_on_ok` is actually `v2_runner_on_ok` with a different method signature.

Also, does not report any changes.

### 1.9

Does not report changes being made.

```
PLAY [example-group] **********************************************************

GATHERING FACTS ***************************************************************
ok: [test-scenario-01]
ok: [test-scenario-02]

TASK: [shell echo 'hi'] *******************************************************
changed: [test-scenario-02]
changed: [test-scenario-01]

TASK: [shell echo 'what'] *****************************************************
changed: [test-scenario-02]
changed: [test-scenario-01]

PLAY RECAP ********************************************************************
shell echo 'what'
test-scenario-01           : ok=3    changed=2    unreachable=0    failed=0
test-scenario-02           : ok=3    changed=2    unreachable=0    failed=0
```

Something is causing `playbook_on_stats` to report the instance methods storing the state as empty.

These issues are somewhat related #437 and #291.

**Comment by retr0h** [comment]
I'd like to ditch this plugin all together.  It's really intended for catching issues when in CI.  The developer would simply need to converge twice on system to find the task which changed.  This is overly complex for no real benefit.

Do others find it useful to know the name of the task which is not idempotent?

**Comment by dj-wasabi** [comment]
I personally don't care to see which task isn't idempotence, that is something you'll need to find out when you are in "developing mode". But I do really see the need for the idempotence check to verify that the module is idempotent. All "my" roles which I develop are idempotent, unless ...

The `idempotence` check just to see if the role is idempotent means a lot for me. 👍 So please keep this running. :-)

**Comment by retr0h** [comment]
> The idempotence check just to see if the role is idempotent means a lot for me. 👍 So please keep this running. :-)

Yeah, I definitely want to keep the idempotence check.  I just don't like the implementation with the callback plugin.

**Comment by retr0h** [comment]
Released molecule [1.10.3](https://pypi.python.org/pypi/molecule) to pypi to address this bug. 

There is no change in workflow.  Molecule still reports if a converge was idempotent or not.  However, it no longer reports which task(s) are not idempotent.  This greatly simplifies the code and issues around this.
